source/Windows.Devices.I2c/I2cController.cs

```diff
@@ -53,6 +53,32 @@ internal static Hashtable DeviceCollection
                 s_deviceCollection = value;
             }
         }
+        /// <summary>
+        /// I2C bus collection associated with this <see cref="I2cController"/>.
+        /// </summary>
+        /// <remarks>
+        /// This collection is for internal use only.
+        /// </remarks>
+        internal static ArrayList BusIdCollection
+        {
+            get
+            {
+                if (s_busIdCollection == null)
+                {
+                    lock (_syncLock)
+                    {
+                        s_busIdCollection = new ArrayList();
```

Comment:
You need to keep the lazy init pattern here. After entering the lock you have to check again if the instance is still null.
In case "this" was the waiting thread, the object could already have been created elsewhere.
Please add the null check here.